pkg/compact/compact.go

```diff
@@ -470,6 +474,70 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	Simulate(ctx context.Context, g *DefaultGrouper, p *largeTotalIndexSizeFilter, sy *Syncer) error
+}
+
+type DefaultPlanSim struct {
+	grouper *DefaultGrouper
+	planner *largeTotalIndexSizeFilter
+	sy      *Syncer
+}
+
+func NewDefaultPlanSim(grouper *DefaultGrouper, planner *largeTotalIndexSizeFilter, sy *Syncer) *DefaultPlanSim {
+	return &DefaultPlanSim{
+		grouper: grouper,
+		planner: planner,
+		sy:      sy,
+	}
+}
+
+func (ps *DefaultPlanSim) Simulate(ctx context.Context) (int, int, error) {
+	numberOfIterations := 0
+	numberOfBlocksToMerge := 0
+
+	if err := ps.sy.SyncMetas(context.Background()); err != nil {
+		return numberOfIterations, numberOfBlocksToMerge, errors.Wrapf(err, "could not sync metas")
+	}
+	originalMetas := ps.sy.Metas()
+
+	// figure out hasPlan and noPlan
+	for {
+		groups, err := ps.grouper.Groups(originalMetas)
```

Comment:
This line should be out of the loop as groups are determined by originalMetas.
originalMetas don't change so groups don't change, too.